src/features/loading.js

```diff
@@ -168,51 +197,50 @@ export const LoadingMixin = (ModelViewerElement) => {
             `${this[$ariaLabel]}. ${this[$ariaLabelCallToAction]}`);
       }
 
-      if (this[$shouldHidePoster]) {
-        posterElement.classList.remove('show');
-        posterElement.setAttribute('aria-hidden', 'true');
-        posterElement.removeAttribute('tabindex');
-      } else {
-        if ((this.preload || this[$dismissPoster]) && this.src) {
-          this[$applyPreloadStrategy]().catch((error) => {
-            this.dispatchEvent(new CustomEvent('error', {detail: error}));
-          });
-        }
-
-        if (this.poster) {
-          posterElement.style.backgroundImage = `url("${this.poster}")`;
-          posterElement.classList.add('show');
-          posterElement.removeAttribute('aria-hidden');
-          posterElement.tabIndex = 1;
-        }
+      if (changedProperties.has('poster') && this.poster) {
+        posterElement.style.backgroundImage = `url("${this.poster}")`;
       }
-    }
 
-    async[$applyPreloadStrategy]() {
-      if (this[$preloadPromise] != null) {
-        return this[$preloadPromise];
+      if (changedProperties.has('src')) {
+        this[$preloadAnnounced] = false;
       }
 
-      if (!this.src) {
-        return;
-      }
+      const preloaded = CachingGLTFLoader.hasFinishedLoading(this.src);
+
+      if (this[$shouldAttemptPreload]) {
+        const detail = {url: this.src};
+
+        if (preloaded) {
+          this.dispatchEvent(new CustomEvent('preload', {detail}));
+        } else {
+          loader.preload(this.src)
```

Comment:
nit: prefer await

Reply:
I'll re-evaluate if this depends on asynchronous code flow

Reply:
I factored this out so that it's wrapped in an async function.